tuf/data/roles.go

```diff
@@ -68,6 +68,16 @@ func ValidRole(name string) bool {
 	return false
 }
 
+// IsBaseRole checks if a role name is valid base role
+func IsBaseRole(name string) bool {
```

Comment:
I'm not sure if I understand exactly what you mean -- like a function that takes a string and outputs the corresponding Role type (or error)?